### ISSUE TITLE
all trends look off - using Binance

### ISSUE THREAD

**Issue by datacryptoz** [Legedric/ptmagic#293]
Dear Community / PT Magic Team,
I hope this is a user error but I am seriously running into issues. I am finding all values incorrect in PT Magic or they feel correct and other times they do not and they look really off.
https://imgur.com/a/qEx4Y
https://youtu.be/YsNCHt48vYA?t=4m35s

The dashboard was never working for me properly as well #'s always look off and I have a lot of 0% values.

To troubleshoot this I have deleted my database multiple times and now only trending the 1h and 24h on Exchange. Please educate or assist.

Thank you,
datacrypto

Here are my settings (I have tried both Relative and Absolute and they both look off. I also tried ExcludeMainCurrency false and true):
{
  "AnalyzerSettings": {
    "MarketAnalyzer": {
      "StoreDataMaxHours": 24, // Number of hours to store market data
      "IntervalMinutes": 1, // Interval in minutes for PTMagic to check market trends and triggers
      "ExcludeMainCurrency": true, // Excludes the main currency (for example BTC) from market trend analysis
      "MarketTrends": [      
	{
          "Name": "Exchange24h", // UNIQUE market trend name (to be referenced by your triggers below)
          "Platform": "Exchange", // Platform to grab prices from (Allowed values are: CoinMarketCap, Exchange)
          "MaxMarkets": 100, // Number of markets/pairs to analyze sorted by 24h volume
          "TrendMinutes": 1440 // Number of minutes to build a trend (1440 = 24h, 720 = 12h, 60 = 1h)
        },
	{
          "Name": "Exchange1h", // UNIQUE market trend name (to be referenced by your triggers below)
          "Platform": "Exchange", // Platform to grab prices from (Allowed values are: CoinMarketCap, Exchange)
          "MaxMarkets": 200, // Number of markets/pairs to analyze sorted by 24h volume
          "TrendMinutes": 60 // Number of minutes to build a trend (1440 = 24h, 720 = 12h, 60 = 1h)
        }
      ]
    },
    "GlobalSettings": [ // Global settings for Profit Trailer properties
      {
        "SettingName": "Default",
        "PairsProperties": {
          "File": "PAIRS.properties"
        },
        "DCAProperties": {
          "File": "DCA.properties"
        },
        "IndicatorsProperties": {
          "File": "INDICATORS.properties"
        }
      }
    ],
    "SingleMarketSettings": [ // Single market/pair settings for Profit Trailer properties
		{
			"SettingName": "1h (-)%1-%1.5",
			"TriggerConnection": "OR",
			"Triggers": [
				{
				"MarketTrendName": "Exchange1h",
				"MarketTrendRelation": "Relative",
				"MaxChange": -1.0,
				"MinChange": -1.5
				}
			],
				"PairsProperties": {
				"ALL_buy_value": -10,
				"ALL_trailing_buy": 0.1
			}
		},
		{ 
			"SettingName": "1h (-)%1.5-%2.5",
			"TriggerConnection": "OR",
			"Triggers": [
				{
				"MarketTrendName": "Exchange1h",
				"MarketTrendRelation": "Relative",
				"MaxChange": -1.5,
				"MinChange": -2.0
				}
			],
				"PairsProperties": {
				"ALL_buy_value": -15,
				"ALL_trailing_buy": 0.2
			}
		},
		{ 
			"SettingName": "1h (-)%2.0-%2.5",
			"TriggerConnection": "OR",
			"Triggers": [
				{
				"MarketTrendName": "Exchange1h",
				"MarketTrendRelation": "Relative",
				"MaxChange": -2.0,
				"MinChange": -2.5
				}
			],
				"PairsProperties": {
				"ALL_buy_value": -20,
				"ALL_trailing_buy": 0.3
			}
		},
		{ 
			"SettingName": "1h (-)%2.5<",
			"TriggerConnection": "OR",
			"Triggers": [
				{
				"MarketTrendName": "Exchange1h",
				"MarketTrendRelation": "Relative",
				"MaxChange": -2.5
				}
			],
				"PairsProperties": {
				"ALL_buy_value": -40,
				"ALL_trailing_buy": 0.3
				}
		}

    ]
  }
}

**Comment by Legedric** [comment]
The screenshots you posted are showing the relative trend for AEBTC.

So for example in this screenshot:
![image](https://user-images.githubusercontent.com/8308141/37198718-8a5c2b70-237f-11e8-8632-118c79d0259c.png)

You can see that the current ABSOLUTE 1h trend for AEBTC is +3.04%  
You can also see that the RELATIVE 1h trend for AEBTC is -2.1%

This means that AEBTC's current 1h trend seems to be 2.1% below the overall average 1h trend of ALL of your markets together.

Without seeing your overall average market trend at that time, I cannot see if the relative trend value there is off or not.

**Comment by datacryptoz** [comment]
Thank you for your quick response,

To help me understand better,
If market trends of all my markets together is for the 1H is 10%+ and in the Dashboard I see:
NANO is +0%
ICX is -10%

Meaning the absolute value of the 2 above coins should be:
NANO +20%
ICX +0%

If that above is true, how is the below possible  (my settings are absolute now on the coin)?

![image](https://user-images.githubusercontent.com/36756792/37239232-973d0ffe-23ec-11e8-963c-75abad788a14.png)

Thank you again. Much appreciated.

**Comment by Legedric** [comment]
Did you try resetting your exchange data?
https://github.com/Legedric/ptmagic/wiki/Reset-PT-Magic-Exchange-Data

**Comment by datacryptoz** [comment]
Yes, I have done this many times. One thing I did notice is that when it builds the market data it only takes a few minutes max on Ubuntu. When I was on Windows it took 10-15 minutes.

And just to be clear is the Dashboard suppose to show relative data or absolute on the coin?

Thank you.

**Comment by Legedric** [comment]
It is showing absolute data.

It should always take 10-15 minutes on Binance to build the initial data. How long did it take exactly on Ubuntu and how many markets (114 BTC markets are all markets at Binance right now) are you running PT with?

**Comment by datacryptoz** [comment]
Hi Legedric,

Right now as I was troubleshooting,

{
"Name": "Exchange24h", //
"Platform": "Exchange", //
"MaxMarkets": 1, // this value was at 100 and 200
"TrendMinutes": 1440 //
}

I have set Market Trends value to 1 for the overall.
This is after I deleted my _data folder.
Here is my Ubuntu load times:

2018/03/10 20:03:33.884 INFO -                          Version 1.2.2
2018/03/10 20:03:33.884 INFO -
2018/03/10 20:03:34.144 INFO - Configuration loaded. Found 5 Market Trends, 1 Gl                                                                                                       obal Settings and 4 Single Market Settings.
2018/03/10 20:03:34.144 INFO - Profit Trailer found
2018/03/10 20:03:34.148 INFO - Trading directory found
2018/03/10 20:03:34.148 INFO - Checking for Pairs Properties file
2018/03/10 20:03:34.148 INFO - PAIRS.properties found!
2018/03/10 20:03:34.148 INFO - Checking for DCA Properties file
2018/03/10 20:03:34.148 INFO - DCA.properties found!
2018/03/10 20:03:34.148 INFO - Checking for Indicators Properties file
2018/03/10 20:03:34.148 INFO - INDICATORS.properties found!
2018/03/10 20:03:34.154 INFO - Checking automated settings for presets...
2018/03/10 20:03:34.154 INFO - Prepared PAIRS.properties for 'Default'
2018/03/10 20:03:34.154 INFO - Prepared DCA.properties for 'Default'
2018/03/10 20:03:34.156 INFO - Prepared INDICATORS.properties for 'Default'
2018/03/10 20:03:34.156 INFO - Checking market trends every 1 minutes...
2018/03/10 20:03:34.194 INFO -
2018/03/10 20:03:34.194 INFO - #################################################                                                                                                       #########
2018/03/10 20:03:34.194 INFO - #************************************************                                                                                                       ********#
2018/03/10 20:03:34.194 INFO - Starting market trend check with Version 1.2.2
2018/03/10 20:03:34.917 INFO - CoinMarketCap - Getting market data...
2018/03/10 20:03:34.958 INFO - CoinMarketCap - Market data received for 100 curr                                                                                                       encies
2018/03/10 20:03:35.107 INFO - CoinMarketCap - 24h market data rebuilt.
2018/03/10 20:03:35.109 INFO - CoinMarketCap - Market data saved.
2018/03/10 20:03:35.109 INFO - CoinMarketCap - Market data cleaned.
2018/03/10 20:03:35.113 INFO - Binance - Getting market data...
2018/03/10 20:03:35.212 INFO - Binance - Market data received for 280 currencies                                                                                                       
2018/03/10 20:03:35.212 INFO - Binance - Getting main market price...
2018/03/10 20:03:35.233 INFO - Binance - Market data received for BTCUSDT
2018/03/10 20:03:35.236 INFO - Binance - Current price for BTCUSDT: 9,075.84 USD                                                                                                       
2018/03/10 20:03:35.302 INFO - Binance - Recreating market data for 114 markets                                                                                                        over 24 hours. This may take a while...
2018/03/10 20:03:36.481 INFO - Binance - No worries, I am still alive... 10/114                                                                                                        markets done...
...
2018/03/10 20:03:47.152 INFO - Binance - No worries, I am still alive... 110/114 markets done...
2018/03/10 20:03:47.505 INFO - Binance - Ticks completed.
2018/03/10 20:03:47.505 INFO - Binance - Creating initial market data ticks. This may take another while...
2018/03/10 20:03:48.260 INFO - Binance - Our nanobots are still at work, hang on... 100/1440 ticks done...
...
2018/03/10 20:03:54.148 INFO - Binance - Our nanobots are still at work, hang on... 1400/1440 ticks 
...
2018/03/10 20:03:55.328 INFO - #******************* RAID SUMMARY ********************#
2018/03/10 20:03:55.328 INFO - + Version: 1.2.2
2018/03/10 20:03:55.328 INFO - + Instance name: PT Magic
2018/03/10 20:03:55.328 INFO - + Time spent: 20s
2018/03/10 20:03:55.328 INFO - + Active setting: Default
2018/03/10 20:03:55.328 INFO - + Global setting changed: No
2018/03/10 20:03:55.328 INFO - + Active single market settings: 13
2018/03/10 20:03:55.328 INFO - + 20 Magicbots killed in 1 raids on Cryptodragon's Lair 1.
2018/03/10 20:03:55.328 INFO -
2018/03/10 20:03:55.328 INFO - DO NOT CLOSE THIS WINDOW! THIS IS THE BOT THAT ANALYZES 


![image](https://user-images.githubusercontent.com/36756792/37246812-5b2a3414-2465-11e8-88da-048ea98972bc.png)

**Comment by smoochy** [comment]
Just as a note, not sure if this is helping at all, but:

I have run PTM on Windows Server 2016 and now on Ubuntu 16.04. Recreating the market data took about 3 minutes per 10 markets for the past 3-5 days (had 5 on Windows, now 3 on Ubuntu).

So I had logfiles like:

> 2018/03/10 20:03:35.302 INFO - Binance - Recreating market data for 114 markets over 96 hours. This may take a while...
> 2018/03/10 20:**06**:36.481 INFO - Binance - No worries, I am still alive... 10/114 markets done...
> 2018/03/10 20:**09**:36.481 INFO - Binance - No worries, I am still alive... 20/114 markets done...

As the recreating of market data in your case is done within seconds, I would assume there is something not working correctly.

**Comment by Nephor80** [comment]
Have the same issue on Binance. I am tracking BTC via "Exchange" and "max. Markets 1" on the 1h, 12h and 24h timeline. 1h seems correct, but there are massive differences on the 12h and 24h frame. Sometimes it seems that 12h and 24h are even changed, although valued correct with correct trendminutes. I am using the "Relative"-setting to apply triggers for each single coin depending on the individual relative change to BTC. Anyway... I deleted the "Exchange"-Folder for a full reset and will try again...

**Comment by Nephor80** [comment]
Anyway... If using only 1 market, should the absolute and relative values be the same at the end?

**Comment by Legedric** [comment]
@Nephor80 No. When you are only using one market in your Profit Trailer (and that's what counts, not the trend) then you will have one absolute number and the relative number will be 0.

**Comment by Legedric** [comment]
@datacryptoz Is the issue still occuring in 1.3.1? I tried to reproduce this but for my trends they look ok so it is somewhat hard to reproduce this issue.

**Comment by Nephor80** [comment]
I had the same issue yesterday, but after deleting the "Exchange"-directory the values now seem correct. Although maybe the 200 markets are too much to calculate, had 75 markets yesterday and values where incorrect, changed to 50 markets and values are correct now. Maybe an issue, maybe not.

**Comment by Eiber** [comment]
Don't know if it's the same issue as previous stated. But how is it possible the relative trend percentage on screen is different from the pairs.properties? For my trends I use MaxMarkets 50. Are these 50 markets also used to calculate the overall market trend, or al ALL markets used?

![relative change](https://user-images.githubusercontent.com/9527985/38138801-fd00c7e0-342c-11e8-9285-ce3df1385b4c.jpg)
